src/runner/test-run/index.js

```diff
@@ -34,6 +36,17 @@ export default class TestRun extends Session {
         this.nativeDialogsInfoTimeStamp = 0;
         this.stepsSharedData            = {};
 
+        this.enableScreenshots = !!this.opts.screenshotPath;
+        this.screenshotCreated = false;
+
+        if (this.enableScreenshots) {
+            if (!this.test.screenshotPath)
```

Comment:
We need store this in `test` because we create a path for screenshot for a test (that can contain several test-runs - for each user agent). We can specify it when the test created but in this case we should use guid-generator for dir naming instead of `Date.now()`, because several tests may be created at the same time